spec/support/features/session_helper.rb

```diff
@@ -351,14 +351,36 @@ def set_up_2fa_with_valid_phone
     end
 
     def register_user(email = 'test@test.com')
+      confirm_email_and_password(email)
+      set_up_2fa_with_valid_phone
+      click_submit_default
+      User.find_with_email(email)
+    end
+
+    def confirm_email_and_password(email)
       allow(FeatureManagement).to receive(:prefill_otp_codes?).and_return(true)
       click_link t('sign_up.registrations.create_account')
       submit_form_with_valid_email(email)
       click_confirmation_link_in_email(email)
       submit_form_with_valid_password
-      set_up_2fa_with_valid_phone
-      click_submit_default
-      User.find_with_email(email)
+    end
+
+    def register_user_with_authenticator_app(email = 'test@test.com')
+      confirm_email_and_password(email)
+      set_up_2fa_with_authenticator_app
+    end
+
+    def set_up_2fa_with_authenticator_app
+      click_link t('links.two_factor_authentication.app_option')
+
+      expect(page).to have_current_path authenticator_setup_path
+
+      secret = page.get_rack_session_key("warden.user.user.session")["new_totp_secret"]
```

Comment:
Why do we have to get the secret from the session? Can't we read it from the page?

Reply:
How? It's not displayed on the page. AFIAK, we can't write a test that simulates adding a code to an authenticator app, or scanning a QR code. Instead, we grab the user's `otp_secret_key` from the session, and we can then use the key to get the current code that would be displayed on an authenticator app.

Reply:
It is displayed on the page though? Isn't this the secret:

![image](https://user-images.githubusercontent.com/963654/38747827-c139241a-3f11-11e8-8e85-0b721e36e23f.png)

I was able to use that with the code in `generate_totp_secret` to sign up just now.

Reply:
Derp! I always use the QR code and didn't connect those dots. I'll fix it.

Reply:
Fixed here: https://github.com/18F/identity-idp/pull/2061/commits/6fc0f2478ad0805b0ebed22354171444a6753b89